razorpay-payments.php

```diff
@@ -210,26 +229,146 @@ public function generate_razorpay_form($orderId)
+
+                    $signUpFee = WC_Subscriptions_Product::get_sign_up_fee($productId);
+
+                    if ($signUpFee)
+                    {
+                        $subscriptionData['addons'] = array(
+                            array (
+                                'item' => array(
+                                    'amount'   => (int) round($signUpFee * 100, 2),
+                                    'currency' => get_woocommerce_currency(),
+                                    'name'     => $product['name']
+                                )
+                            )
+                        );
+                    }
+
+                    $api = $this->getRazorpayApiInstance();
+
+                    $subscription = $api->subscription->create($subscriptionData);
+                    $subscriptionId = $subscription['id'];
+                }
+            }
+
             $redirectUrl = get_site_url() . '/?wc-api=' . get_class($this);
 
             $razorpayOrderId = $this->createOrGetRazorpayOrderId($orderId);
 
-            if ($razorpayOrderId === null)
+            if (($razorpayOrderId === null) and
+                (wcs_order_contains_subscription($orderId) === false))
             {
                 return 'RAZORPAY ERROR: Api could not be reached';
             }
 
-            $checkoutArgs = $this->getCheckoutArguments($order, $razorpayOrderId);
+            $checkoutArgs = $this->getCheckoutArguments($order, $razorpayOrderId, $subscriptionId);
 
             $html = $this->generateOrderForm($redirectUrl, $checkoutArgs);
 
             return $html;
         }
 
+        protected function getPlanArguments($product)
+        {
+            $productId = $product['product_id'];
+
+            $period       = WC_Subscriptions_Product::get_period($productId);
+            $interval     = WC_Subscriptions_Product::get_interval($productId);
+            $recurringFee = WC_Subscriptions_Product::get_price($productId);
+
+            $planArgs = array(
+                'item' => array(
+                    'name'     => $product['name'],
+                    'amount'   => (int) round($recurringFee * 100, 2),
+                    'currency' => get_woocommerce_currency(),
+                ),
+                'period'   => $this->getProductPeriod($period),
+                'interval' => $interval
+            );
+
+            return $planArgs;
+        }
+
+        protected function getProductPeriod($period)
+        {
+            $periodMap = array(
+                'day'   => 'daily',
+                'week'  => 'weekly',
+                'month' => 'monthly',
+                'year'  => 'yearly'
+            );
+
+            return $periodMap[$period];
+        }
+
+        protected function getCustomerId($order)
+        {
+            $api = $this->getRazorpayApiInstance();
+
+            $data = array(
+                'name'          => $order->billing_first_name." ".$order->billing_last_name,
```

Comment:
not compatible with 3.0

See https://github.com/razorpay/razorpay-woocommerce/pull/35